### PR TITLE
Reader: move reader/share to blocks

### PR DESCRIPTION
**Pull request by bluefuton**
Removes the fairly ancient `reader/share` component to the blocks directory, ES6-ifies it, and removes _some_ usage of the now-deprecated SitesList. I'd originally intended to remove all uses of SiteList but it's not currently possible (see https://github.com/Automattic/wp-calypso/pull/12779#issuecomment-292597978).

### CODE REVIEW

**Comment by blowery** [comment]
does anything still use it?

**Comment by bluefuton** [comment]
Still todo:
- [ ] migrate CSS
- [ ] replace main use of `SitesList` in `<SitePopover>` - @blowery do you know what we need to swap in here?
- [ ] fix popover position in devdocs example

**Comment by bluefuton** [comment]
@gwwar do you know if it's currently possible to replace SitesList here? We're currently handing it to `SitesPopover` but I can't see an obvious Redux replacement for it.

https://github.com/Automattic/wp-calypso/pull/12779/files#diff-44664728af70e4ff1cefb3a384a52086R195

**Comment by gwwar** [comment]
@bluefuton looks like SitesPopover is passing through a sites list instance for SiteSelector. There's no avoiding this atm, but folks are planning on ripping out SitesList more aggressively later this month.

**Comment by bluefuton** [comment]
Thanks @gwaar! I'll leave that last use of sites list in there for the moment then.

File: client/blocks/reader-share/index.jsx [hunk 1]
```diff
@@ -1,23 +1,37 @@
...
+
+/**
+ * Local variables
+ */
+// Remove me
+const sitesList = require( 'lib/sites-list' )(); // eslint-disable-line no-restricted-modules
+
+const user = User();
+const actionMap = {
+	twitter( post ) {
```
Comment (samouri) [inline client/blocks/reader-share/index.jsx]: i didn't realize this was valid syntax!

File: client/blocks/reader-share/index.jsx [hunk 2]
```diff
@@ -70,73 +84,75 @@ function buildQuerystringForPost( post ) {
...
+	static propTypes = {
 		iconSize: React.PropTypes.number
-	},
+	}
 
-	getInitialState() {
-		return { showingMenu: false };
-	},
+	static defaultProps = {
+		position: 'bottom',
```
Comment (samouri) [inline client/blocks/reader-share/index.jsx]: did you change the defaultProp for `postition` from top --> bottom on purpose?

File: client/blocks/reader-share/index.jsx [hunk 3]
```diff
@@ -70,73 +84,75 @@ function buildQuerystringForPost( post ) {
...
+			const target = canShareToWordPress() ? 'wordpress' : 'external';
 			stats.recordAction( 'open_share' );
 			stats.recordGaEvent( 'Opened Share to ' + target );
 			stats.recordTrack( 'calypso_reader_share_opened', {
 				target
 			} );
 		}
-		this._deferMenuChange( ! this.state.showingMenu );
-	},
+		this.deferMenuChange( ! this.state.showingMenu );
```
Comment (samouri) [inline client/blocks/reader-share/index.jsx]: why does `deferMenuChange` (and the closeHandle) logic exist? 
As opposed to directly calling setState here.

File: client/blocks/reader-share/index.jsx (hunk 2 above)
Comment (bluefuton) [inline client/blocks/reader-share/index.jsx]: Yep. We need `bottom` in both places it's used (bottom of post card and devdocs) so it seemed like a more sensible default.

File: client/blocks/reader-share/index.jsx (hunk 3 above)
Comment (bluefuton) [inline client/blocks/reader-share/index.jsx]: There are some comments about that in `closeMenu`:

```
// have to defer this to let the mouseup / click escape.
// If we don't defer and remove the DOM node on this turn of the event loop,
// Chrome (at least) will not fire the click
```

https://github.com/Automattic/wp-calypso/blob/36866253676a12eb2769f2187eb1933c551dbb4d/client/blocks/reader-share/index.jsx#L139

**Comment by samouri** [comment]
Should https://github.com/Automattic/wp-calypso/pull/12779/files#diff-44664728af70e4ff1cefb3a384a52086R73 have both an image and an embed or do we usually want one or the other?

File: client/blocks/reader-share/index.jsx [hunk 4]
```diff
@@ -70,73 +84,75 @@ function buildQuerystringForPost( post ) {
...
+	}
 
-	closeMenu() {
+	closeMenu = () => {
 		// have to defer this to let the mouseup / click escape.
 		// If we don't defer and remove the DOM node on this turn of the event loop,
 		// Chrome (at least) will not fire the click
 		if ( this.isMounted() ) {
```
Comment (samouri) [inline client/blocks/reader-share/index.jsx]: using `isMounted` is deprecated and produces a React warning in the console.

screen shot 2017-04-10 at 10 56 15 am

**Comment by samouri** [comment]
Sharing to FB/Twitter seems to be broken for me on this branch whereas it still works in prod

File: client/blocks/reader-share/index.jsx (hunk 4 above)
Comment (bluefuton) [inline client/blocks/reader-share/index.jsx]: @blowery as the original author here, do you know if we'll have to keep track of mounted state ourselves?

**Comment by bluefuton** [comment]
Now keeping track of mounted state internally rather than using isMounted(), which has fixed the FB/Twitter share too @samouri.

**Comment by samouri** [comment]
giving it another test drive

**Pull request merged**